naga/src/front/wgsl/parse/mod.rs

```diff
@@ -2750,6 +2771,8 @@ impl Parser {
         lexer: &mut Lexer<'a>,
         out: &mut ast::TranslationUnit<'a>,
     ) -> Result<'a, ()> {
+        let doc_comments = lexer.accumulate_doc_comments();
```

Comment:
This approach is interesting, as we could use it to emit warning when doc comments are encountered when it shouldn't have been: check that `doc_comments` is empty for items that don't support it (separator, word...).

I'd consider this for another PR though, as it will lead to questions about error reporting, etc...

Reply:
I guess we could just do it based on `options.parse_doc_comments`.